cryptoki/src/functions/slot_token_management.rs

```diff
@@ -108,6 +109,50 @@ impl Pkcs11 {
             Ok(TokenInfo::new(token_info))
         }
     }
+
+    /// Get all mechanisms support by a slot
+    pub fn get_mechanisms(&self, slot: Slot) -> Result<Vec<MechanismType>> {
+        let mut mechanism_count = 0;
+
+        unsafe {
+            Rv::from(get_pkcs11!(self, C_GetMechanismList)(
+                slot.into(),
+                std::ptr::null_mut(),
+                &mut mechanism_count,
+            ))
+            .into_result()?;
+        }
+
+        let mut mechanisms = vec![0; mechanism_count.try_into()?];
+
+        unsafe {
+            Rv::from(get_pkcs11!(self, C_GetMechanismList)(
+                slot.into(),
+                mechanisms.as_mut_ptr(),
+                &mut mechanism_count,
+            ))
+            .into_result()?;
+        }
+
+        Ok(mechanisms
+            .into_iter()
+            .filter_map(|type_| type_.try_into().ok())
+            .collect())
```

Comment:
Is this not similar to the other issues we've had where maybe we don't support all the mechanism types that the token offers, but we don't want this operation to fail because of it?

Reply:
> Is this not similar to the other issues we've had where maybe we don't support all the mechanism types that the token offers, but we don't want this operation to fail because of it?

Yes, considering that the only failure here is an unsupported mechanism, I believe it's safe to ignore.

Reply:
Right I see your points, makes sense. And we even log it as an `error` inside the `TryFrom` trait implementation so it will be easier to debug.